clients/drcachesim/analyzer.h

```diff
@@ -120,18 +121,24 @@ template <typename RecordType, typename ReaderType> class analyzer_tmpl_t {
         analyzer_worker_data_t(int index, typename sched_type_t::stream_t *stream)
             : index(index)
             , stream(stream)
+            , cur_interval_index(0)
+            , shard_data()
         {
         }
         analyzer_worker_data_t(analyzer_worker_data_t &&src)
         {
             index = src.index;
             stream = src.stream;
+            cur_interval_index = src.cur_interval_index;
+            shard_data = std::move(src.shard_data);
             error = std::move(src.error);
         }
 
         int index;
         typename scheduler_tmpl_t<RecordType, ReaderType>::stream_t *stream;
         std::string error;
+        uint64_t cur_interval_index;
```

Comment:
This needs to be per-shard, not per-worker, as the scheduler does not guarantee to run each shard to completion before switching to another shard on that worker.

Reply:
Done.